EasyReflectometry/sample/elements/layers/layer_area_per_molecule.py

```diff
@@ -87,43 +87,44 @@ def __init__(
         :param name: Name of the layer, defaults to "EasyLayerAreaPerMolecule"
         :param interface: Interface object, defaults to `None`
         """
-        scattering_length = neutron_scattering_length(molecular_formula)
+
+        # Create the solvated molecule and corresponding constraints
+        molecule = Material.from_pars(0.0, 0.0, name=molecular_formula, interface=interface)
+
         default_options = deepcopy(LAYER_AREA_PER_MOLECULE_DETAILS)
         del default_options['sl']['value']
         del default_options['isl']['value']
-        scattering_length_real = Parameter('scattering_length_real', scattering_length.real, **default_options['sl'])
-        scattering_length_imag = Parameter('scattering_length_imag', scattering_length.imag, **default_options['isl'])
-        sld_real = area_per_molecule_to_sld(scattering_length_real.raw_value, thickness.raw_value, area_per_molecule.raw_value)
```

Comment:
What is sld? :)  Is it the same story as APM? Can it be replaced with a long variable name?